pkg/reconciler/pipelinerun/resources/apply.go

```diff
@@ -177,9 +177,8 @@ func ApplyTaskResults(targets PipelineRunState, resolvedResultRefs ResolvedResul
 			pipelineTask := resolvedPipelineRunTask.PipelineTask.DeepCopy()
 			pipelineTask.Params = pipelineTask.Params.ReplaceVariables(stringReplacements, arrayReplacements, objectReplacements)
 			if pipelineTask.IsMatrixed() {
-				// Only string replacements from string, array or object results are supported
-				// We plan to support array replacements from array results soon (#5925)
-				pipelineTask.Matrix.Params = pipelineTask.Matrix.Params.ReplaceVariables(stringReplacements, nil, nil)
+				// String replacements from string, array or object results or array replacements from array results are supported
```

Comment:
This might be a little confusing. how about something like: `results replacements support: 1. string replacements from string, array or object results 2. array replacements from array results` ?